File: src/jionchu/Week1/위장.kt
/* Programmers
 * 해시 - 위장
 * created on 2021.09.16
 * created by jionchu */

class Solution {
    fun solution(clothes: Array<Array<String>>): Int {
        var answer = 1
        
        clothes.groupBy { it[1] }.forEach { answer *= (it.value.size + 1) }
        
        return answer - 1
    }
}

File: src/kyoungjin/Week1/위장.js
function solution(clothes) {
  let answer = 1;

  const table = clothes.reduce((acc, cur) => {
    acc[cur[1]] = acc[cur[1]] ? acc[cur[1]] + 1 : 1;
    return acc;
  }, {});

  for (let key in table) {
    answer *= table[key] + 1; // 아무 것도 안입은 경우 추가
  }

  return answer - 1; // 전부 안입은 경우는 빼기
}
